lisa/sut_orchestrator/azure/features.py

```diff
@@ -1120,7 +1120,34 @@ def _do_update_subnet(
         )
         self._log.debug(f"Checking subnet: {subnet_az.address_prefix} == {subnet_mask}")
         # Step 4: once we find the matching subnet, assign the routing table to it.
-        if subnet_az.address_prefix == subnet_mask:
+        if subnet_az.address_prefix is not None:
+            az_subnet = subnet_az.address_prefix
```

Comment:
Please add comments above each logic block to improve understanding of the new code.